plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -530,9 +532,25 @@ outOfBoundsErr x branches = fold
     ]
 
 instance AnnotateCaseBuiltin DefaultUni where
-    annotateCaseBuiltin (SomeTypeIn uni) branches = case uni of
+    annotateCaseBuiltin ann (SomeTypeIn uni) branches = case uni of
         DefaultUniBool    -> Right $ map (, []) branches
         DefaultUniInteger -> Right $ map (, []) branches
+        DefaultUniList ty ->
+          case branches of
+            [cons] ->
+              Right
+                [ (cons, [ TyBuiltin @TyName ann $ SomeTypeIn ty
+                         , TyBuiltin @TyName ann $ SomeTypeIn $ DefaultUniList ty
```

Comment:
You can save some typing with `PlutusCore.MkPlc.mkTyBuiltinOf`.

You shouldn't need the `TyName` type application I think?